Configure

```diff
@@ -1321,6 +1320,10 @@ if (defined($config{api})) {
     push @{$config{defines}}, $apiflag;
 }
 
+if (defined($predefined{__clang__}) && !$disabled{asm}) {
```

Comment:
Why is the status of asm enablement relevant here?

Reply:
Because it complains most when compiling .s modules. It's complaining about not having use -DTHIS_N_THAT since it doesn't pre-process .s files.

Reply:
I feel like we might as well just always silence the unused-arguments warnings; look at things like ```ssl/statem/extensions*.c``` -- we have various function pointers in the extensions_definition_st, and many of the per-extension implementations thereof have no use for some of the arguments.  That's just part of the way we designed the table interface, and we are choosing a design that includes unused arguments in common cases, so the warning is just noise, there.

Reply:
Or wait, I am thinking of -Wunused-parameter, never mind.